### PR TITLE
add test for multi exposure dialog

### PR DESCRIPTION
**Pull request by Gustry**
### What does it fix?
* Ticket: #
* Funded by: DFAT
* Description: add test for multi exposure dialog

### Checklist:
- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Add to the changelog in metadata.txt if it's a new feature
- [ ] Unit test for new code added
- [ ] Request someone to review or test your PR

### CODE REVIEW

**Comment by Gustry** [comment]
@ismailsunni What I doing wrong, there is not layer in my iface. Test is working very well in pycharm, but not in docker. I already tried to move my get_qgis_app, etc but it seems I'm missing something. Thanks

**Comment by ismailsunni** [comment]
Ah, I remember this. I have the issue before. Basically, the layers are not added to map canvas for some reason. No idea why, but IIRC one of QGIS dev told me that it has different signal or something. I couldn't remember it.
Anyway, to fix it, just add the layers manually to map canvas.

I made a PR to this branch https://github.com/Gustry/inasafe/pull/11

See also the comment that I put to see my old problem.

**Comment by Gustry** [comment]
Ok thanks for the fix.

I took the same pattern as `test_multi_buffer.py` and it's working well.

**Comment by ismailsunni** [comment]
> I took the same pattern as test_multi_buffer.py and it's working well.

Perhaps because it use QgsMapLayerComboBox? Not sure though.

**Comment by Gustry** [comment]
Ok, I'm giving another try to get rid of the internal fake iface.

I'm adding a new function to start the QGIS app.
* Use the mock iface by QGIS
* Use the QGIS App from QGIS
* Put python imports always before python code (not like the get_qgis_app before import)
* Migration to QGIS 3 will be easy

Can you have a look to this PR @ismailsunni and @timlinux?
Then if, try to use this new function instead of the get_qgis_app. Then if nothing is wrong during a few weeks, we can start migrating files after files. Is-it ok?

File: safe/gui/tools/multi_exposure_dialog.py [hunk 1]
```diff
@@ -316,11 +316,7 @@ def _populate_reporting_tab(self):
             if flag and loaded_layer not in canvas_layers:
                 continue
 
-            if qgis_version() >= 21800:
-                title = loaded_layer.name()
-            else:
-                # QGIS 2.14
-                title = loaded_layer.layerName()
+            title = loaded_layer.name()
```
[the remark this answers is not in the document]
Comment (Gustry) [reply safe/gui/tools/multi_exposure_dialog.py]: No, the API is like this in 2.14, it was a mistake before.

BTW, 2.14 has reached the end of life, no more new release of 2.14 now according to the roadmap.

**Pull request merged**